js/src/javascript/beautifier.js

```diff
@@ -185,6 +185,7 @@ Beautifier.prototype.create_flags = function(flags_base, mode) {
     inline_frame: false,
     if_block: false,
     else_block: false,
+    class_start_block: false, // class A { INSIDE HERE } or class B extends C { INSIDE HERE }
```

Comment:
Sorry about the multiple review passes. 

Should this be called just `class_block`?

I'm thinking there should be a `class_definition` flag as well - set when the  `class` reserved word is seen. Then that continues until a `{` is seen, at which point `class_definition` is cleared and `class_block` is set. This an example of where "look back" works, but there's enough information, that it isn't needed.

Do you see what I'm saying?